### ISSUE TITLE
Copy before truncate not working.

### ISSUE THREAD

**Issue by owilsonjunior** [logrotate/logrotate#591]
Hi guys. 
First of all, I apologize for the bad English...

I need some support in rotating the logs.

I use the configuration below for the standard rotation of the Tomcats I implement. 

_/opt/tomcat_planodesaude/logs/*.out  /opt/tomcat_planodesaude/logs/*.log {
    su tomcat tomcat
    missingok
    copytruncate
    daily
    rotate 7
}_

For some time now, I've noticed a lot of data loss in the rotation that is created. 
It's as if it only manages to rotate from the time the task starts until it finishes, not saving the log data from the last 24 hours as expected: 

![1](https://github.com/logrotate/logrotate/assets/50386637/f8a079bb-8e22-43a0-85ea-0f07c3dbd7fb)

![2](https://github.com/logrotate/logrotate/assets/50386637/b2a0271f-4ea3-4f36-bc18-5289b96f5b0e)

Note that the start date of the "copy" created and the end date are a little over 15 minutes apart. 

Is there any way to ensure that it creates a copy of the data from the last 24 hours?

**Comment by cgzones** [comment]
Maybe the problem is tomcat after the rotation still writing to the old offset; it might need a reload.
Does the log file contain more content than being saved just before the rotation?

**Comment by owilsonjunior** [comment]
No, only that time interval and in the original file we have the continuation from the moment of truncate.

**Comment by owilsonjunior** [comment]
Can we get around this?
